src/index.ts

```diff
@@ -1,69 +1,290 @@
 import {
   JupyterFrontEnd,
-  JupyterFrontEndPlugin
+  JupyterFrontEndPlugin,
+  ILayoutRestorer
 } from '@jupyterlab/application';
 import { deployAppIcon } from './icons';
-import { DocumentWidget } from '@jupyterlab/docregistry';
+import { WidgetTracker, IFrame, MainAreaWidget } from '@jupyterlab/apputils';
+import { Widget } from '@lumino/widgets';
+
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+
+const DEFAULT_BASE_URL = '/services/japps/create-app';
+
+const PLUGIN_ID = 'jupyterlab-jhub-apps:commands';
+
+/**
+ * Settings for the runtime configuration of the application.
+ * These are read from the settings registry and updated any
+ * time a user changes their settings.
+ */
+const runtimeSettings = {
+  queryParameters: {} as {
+    headless?: string;
+    filepath?: string;
+  },
+  baseUrl: DEFAULT_BASE_URL
+};
 
 namespace CommandIDs {
   /**
-   * Opens the URL to deploy the application with pre-populated fields
+   * Opens the URL for deploying an app with pre-populated fields.
    */
   export const deployApp = 'jhub-apps:deploy-app';
 }
 
-interface IDeployAppArgs {
+interface IDeployAppBaseConfig {
   /**
-   * The origin of the command e.g. main-menu, context-menu, etc.
+   * The base URL of the app deployment form.
+   */
+  baseUrl?: string;
+  /**
+   * Additional query parameters to be appended to the baseUrl.
+   */
+  queryParameters?: {
+    headless?: string;
+    filepath?: string;
+  };
+}
+
+interface IDeployAppConfig extends IDeployAppBaseConfig {
+  /**
+   * The origin of the command e.g. main-menu, context-menu, etc.,
+   * which determines the icon and label to display.
    */
   origin?: string;
+  IShell?: JupyterFrontEnd.IShell;
+  widgetTracker?: WidgetTracker<MainAreaWidget<DeployAppWidget>>;
+}
+
+interface IDeployAppWidgetArgs extends IDeployAppBaseConfig {}
+
+interface IPathWidget {
+  /**
+   * The context object associated with the widget.
+   * Contains the metadata and information about the widget.
+   */
+  context: {
+    /**
+     * The path of the current document or resource associated with the widget.
+     * Typically represents the file path or location of the notebook or document.
+     */
+    path: string;
+  };
+}
+
+function applySettings(setting: ISettingRegistry.ISettings): void {
```

Comment:
This is the callback used by `settings.changed.connect`. It closes over the global state `runtimeSettings` and updates these values.